libraries/AP_ICEngine/AP_ICEngine.cpp

```diff
@@ -397,8 +405,14 @@ void AP_ICEngine::update(void)
         if (!should_run) {
             state = ICE_OFF;
         } else if (now - starter_last_run_ms >= starter_delay*1000) {
-            GCS_SEND_TEXT(MAV_SEVERITY_INFO, "Starting engine");
-            state = ICE_STARTING;
+            // check if we should retry starting the engine
+            if (max_crank_retry >= 0 && (max_crank_retry == 0 || crank_retry_ct < max_crank_retry)) {
```

Comment:
Anything less than zero should be considered infinite retry. As written, -1 means you can't start the engine ever.

Should be
`if (max_crank_retry > 0 && crank_retry_ct < max_crank_retry) {`

Reply:
if (max_crank_retry <= 0 || crank_retry_ct < max_crank_retry) 
   
   this will be a fix as we want it to try infinitely when it is 0 or less, or less then maxcount.